pepper-apis/docs/specification/src/components/responses.yml

```diff
@@ -55,6 +55,13 @@ ActivityAnswersPatchResponse:
             description: list of updated block visibilities
             items:
               $ref: '../pepper.yml#/components/schemas/Block.Visibility'
+          compositeVisibility:
+            type: array
+            nullable: true
+            description: |
+              list of visibility updates for composite questions
+            items:
+              $ref: '../pepper.yml#/components/schema/Composite.Visibility'
```

Comment:
Did we consider introducing blocks for composite children? Good part would be that we would not require a separate structure to  keep track of composite children in JSON; it would just be another block with an id, so `compositeVisibility` could go away and we could just stick with `blockVisibility`. The bad part would be that we would have to have blocks for children. Not sure how hard or how much work this would be though. Did we look at blocks for composite children? @bskinner @ssettipalli

Reply:
We did- this is a major api change and far outside our normal use since it inverts the hierarchy (as it would be a question containing blocks). It still does not solve the identification issue since just the block id would not uniquely identify a row in a composite response, and blockVisibility would need to be modified to support the potential for multiple "rows", which would be unusual in the context of the rest of the API.